src/ui/components/shared/MaterialIcon.tsx

```diff
@@ -18,6 +18,22 @@ type MaterialIconProps = React.HTMLProps<HTMLDivElement> & {
   iconSize?: keyof typeof SIZE_STYLES;
 };
 
+let isChecking = false;
+function useMaterialIconCheck() {
+  useEffect(() => {
+    if (isChecking) {
+      return;
+    }
+    isChecking = true;
+    let id = setInterval(() => {
```

Comment:
`isChecking` and a handle for interval always go in pair, so `isChecking` is redundant. We can do something like `const intervalHandleId: number = useRef(-1)` and just use that to check things.